### PR TITLE
add route for get account history

### PR DESCRIPTION
**Pull request by AlisherNurlybaev**

### CODE REVIEW

File: utilities/requests/hiveEngineRequests.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const axiosRetry = require('axios-retry');
+const axios = require('axios');
+
+module.exports = async (params) => {
+  axiosRetry(axios, {
+    retries: 3,
+    retryDelay: (retryCount) => retryCount * 100,
+    retryCondition: (error) => error.response.status !== 200,
+  });
```
Comment (Fesmofet) [inline utilities/requests/hiveEngineRequests.js]: can it impact on whole axios  in our app? 
maybe better create an instance?

File: utilities/requests/hiveEngineRequests.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  const res = await axios.get('https://accounts.hive-engine.com/accountHistory', { params })
+    .then((response) => response)
+    .catch((error) => error);
+
+  return res;
```
Comment (Fesmofet) [inline utilities/requests/hiveEngineRequests.js]: we have try catch approach in whole app, let's continue to do same way

File: models/engineAccountHistoryModel.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const { EngineAccountHistories } = require('database').models;
+
+exports.find = async ({ condition, skip, limit }) => {
+  try {
+    return {
+      result: await EngineAccountHistories
+        .find(condition)
+        .sort({ timestamp: -1 })
```
Comment (Waiviogit) [inline models/engineAccountHistoryModel.js]: why hardcode?

File: routes/routes.js [hunk 4]
```diff
@@ -7,21 +7,27 @@ const {
...
 const mailerRoutes = new Router();
 const apiRoutes = new Router();
+const hiveEngineRoutes = new Router();
+
 
 apiRoutes.use('/campaigns-api/referrals', referralRoutes);
 apiRoutes.use('/campaigns-api/withdraw', withdrawRoutes);
 apiRoutes.use('/campaigns-api/mailer', mailerRoutes);
 apiRoutes.use('/campaigns-api', campaignsRoutes);
+apiRoutes.use('/campaigns-api', accountRoutes);
```
Comment (Waiviogit) [inline routes/routes.js]: do you need this?

File: swagger/swagger.json [hunk 5]
```diff
@@ -176,6 +176,86 @@
...
+            "in": "body",
+            "type": "array",
+            "required": false
+          },{
+            "name": "timestampEnd",
+            "in": "body",
+            "type": "number",
+            "required": false
+          }
+        ],
```
Comment (Waiviogit) [inline swagger/swagger.json]: wrong docs(body params)
Example
/campaigns-api/api-docs/#/Campaign/post_campaigns_reserved

File: utilities/operations/account/getHistory.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    if (error) return { error };
+
+    const resArray = _.concat(sortedRes, result).sort((x, y) => y.timestamp - x.timestamp);
+
+    const updateSkip = resArray.indexOf(_.find(resArray, (obj) => obj.timestamp === params.timestampEnd)) + 1;
+
+    const history = resArray.slice(updateSkip, updateSkip + params.limit);
+
+    return { history };
+  } else {
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: why do you need else if you return history from if?

File: utilities/operations/account/getHistory.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  let operator = '$or';
+
+  if (params.excludeSymbols) {
+    condition = { $nin: params.excludeSymbols };
+    operator = '$and';
+  }
+
+  if (params.timestampEnd) {
+    const data = {
+      symbol: params.symbol,
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: what would be if params.symbol  - undefined?

File: utilities/operations/account/getHistory.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+      account: params.account,
+      offset: 0,
+      limit: params.limit,
+      timestampEnd: params.timestampEnd,
+      timestampStart: 1,
+    };
+
+    const res = await hiveEngineRequests(data);
+    const sortedRes = _.filter(res.data, (el) => !_.includes(params.excludeSymbols, el.symbol));
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: what will be if limit 20 and 20 latest records is excluded symbols

File: utilities/operations/account/getHistory.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+    const { result, error } = await engineAccountHistoryModel.find({
+      condition: {
+        account: params.account,
+        timestamp: { $lte: params.timestampEnd },
+        [operator]: [{ symbol: condition }, { symbolOut: condition }, { symbolIn: condition }],
+      },
+    });
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: duplicate code fragment you can just add method that will create params for request

File: utilities/operations/account/hiveEngineRequests.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+  axiosRetry(instance, {
+    retries: 3,
+    retryDelay: (retryCount) => retryCount * 100,
+    retryCondition: (error) => error.response.status !== 200,
+  });
+
+  return await instance.get('https://accounts.hive-engine.com/accountHistory', { params })
+    .then((response) => response)
+    .catch((error) => error);
+};
```
Comment (Waiviogit) [inline utilities/operations/account/hiveEngineRequests.js]: replace to hiveEngine/engineOPerations.js

merge dev in current branch!

File: utilities/operations/account/getHistory.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+    const sortedRes = _.filter(res.data, (el) => !_.includes(params.excludeSymbols, el.symbol));
+    const {
+      result,
+      error
+    } = await engineAccountHistoryModel.find({
+      condition: {
+        account: params.account,
+        [operator]: [{ symbol: condition }, { symbolOut: condition }, { symbolIn: condition }],
+      },
+      skip: 0,
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: why do you need send this param "skip: 0"

File: utilities/operations/account/hiveEngineRequests.js [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+
+  axiosRetry(instance, {
+    retries: 3,
+    retryDelay: (retryCount) => retryCount * 100,
+    retryCondition: (error) => error.response.status !== 200,
+  });
+
+  return await instance.get('https://accounts.hive-engine.com/accountHistory', { params })
+    .then((response) => response)
+    .catch((error) => error);
```
Comment (Waiviogit) [inline utilities/operations/account/hiveEngineRequests.js]: try catch please

File: utilities/operations/account/getHistory.js [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+        timestamp: { $lte: params.timestampEnd },
+        [operator]: [{ symbol: condition }, { symbolOut: condition }, { symbolIn: condition }],
+      },
+    });
+
+    if (error) return { error };
+
+    const resArray = _.concat(sortedRes, result).sort((x, y) => y.timestamp - x.timestamp);
+
+    const updateSkip = resArray.indexOf(_.find(resArray, (obj) => obj.timestamp === params.timestampEnd)) + 1;
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: why+1?

File: utilities/operations/account/getHistory.js [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+  if (params.excludeSymbols) {
+    condition = { $nin: params.excludeSymbols };
+    operator = '$and';
+  }
+
+  if (params.timestampEnd) {
+    const data = {
+      symbol: params.symbol,
+      account: params.account,
+      offset: 0,
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: offset is optional why do you need it?

File: utilities/operations/account/getHistory.js [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+
+    const updateSkip = resArray.indexOf(_.find(resArray, (obj) => obj.timestamp === params.timestampEnd)) + 1;
+
+    const history = resArray.slice(updateSkip, updateSkip + params.limit);
+
+    return { history };
+  } else {
+
+    const data = {
+      symbol: params.symbol,
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: same(can be undefined)

File: utilities/operations/account/getHistory.js [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+      account: params.account,
+      offset: 0,
```
Comment (Waiviogit) [inline utilities/operations/account/getHistory.js]: optional

**Pull request merged**